### PR TITLE
3 Instagram API homework #2 - Aleks Gechovski

### PR DESCRIPTION
**Pull request by AlexGechovski**

### CODE REVIEW

File: homeworks/3/demo/src/main/java/com/example/demo/Login.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.sql.*;
+
+@WebServlet(name = "login", value = "/login")
+public class Login extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+    }
```
Comment (allks12) [inline homeworks/3/demo/src/main/java/com/example/demo/Login.java:18]: I recommend you do not override this method since now when the user calls the GET method it won't do anything, and they will wonder what is going on. If the method is not present in the code, the user will get an error message and know that this operation is not valid.

File: homeworks/3/demo/src/main/java/com/example/demo/Register.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+
+@WebServlet(name = "Register", value = "/register")
+public class Register extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+    }
```
Comment (allks12) [inline homeworks/3/demo/src/main/java/com/example/demo/Register.java:19]: I recommend you do not override this method since now when the user calls the GET method it won't do anything, and they will wonder what is going on. If the method is not present in the code, the user will get an error message and know that this operation is not valid.

File: homeworks/3/demo/src/main/java/com/example/demo/posts.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            System.err.println(e.getMessage());
+        }
+
+        Gson gson = new Gson();
+        Post post = gson.fromJson(jb.toString(),Post.class);
+
+
+
+
+
```
Comment (allks12) [inline homeworks/3/demo/src/main/java/com/example/demo/posts.java:47]: There is no need for such big gaps of empty lines. They make your code longer and harder to read.

**Review by allks12 — COMMENTED** [review]
Overall I can say you did your homework excellently. You have the full package of functionalities, used response codes, sessions and JSON and you tried using data base. I would recommend minor changes such as not overriding the doGet methods where not needed and giving messages to the user when you give them response codes (this last part is not necessary but will be great in regard of user experience), deleting unnecessary empty lines in your code (this will make it easier to read) and using more types of response codes so the user gets a clearer idea of what is going on.
In conclusion of the above statements, I would recommend as your grade 5.75.
